### ISSUE TITLE
Face Liveness in React Native

### ISSUE THREAD

**Issue by finexusCriss** [aws-amplify/amplify-ui#4636]
### On which framework/platform would you like to see this feature implemented?

React Native

### Which UI component is this feature-request for?

Liveness

### Please describe your feature-request in detail.

AWS Face Liveness in Swift is really good to use, however, I would like to implement it in my React Native project, but the bridging part is not so direct to do, will be really great if there's a ready to use React Native SDK for the Face Liveness too

### We love contributors! Is this something you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request.
- [ ] ⚠️ This feature might incur a breaking change.

**Comment by ayush987goyal** [comment]
Hi @finexusCriss , thanks for the trying out Face Liveness components.

The Amplify Face Liveness Android (Kotlin) and iOS (Swift) SDKs can be integrated into ReactNative application by using [Native-Modules](https://reactnative.dev/docs/native-modules-intro) . With Native modules, you can start a new activity (in Android) or a `UIViewController` (in iOS) to render the Amplify Liveness components.

**Comment by finexusCriss** [comment]
Hi @ayush987goyal ,thank you so much for the prompt reply, now I feel safe working towards the use of Native-Modules,  previously was not sure if it work this way

**Comment by reesscot** [comment]
@finexusCriss Thank you for your feature request! While it is possible to integrate the Android / iOS SDK's into a React Native application, please note that this is not officially supported.

We will consider adding an official React Native component to our roadmap.

**Comment by finexusCriss** [comment]
Now I'm writing a React Native wrapper for the iOS part of AWS Face Liveness (I'm trying to make the AWS FaceLiveness available as a React Native library), 
however the import for FaceLiveness is giving `No such module 'FaceLiveness'`,
<img width="1187" alt="Screenshot 2023-11-08 at 10 53 37 PM" src="https://github.com/aws-amplify/amplify-ui/assets/106042257/086d1430-13f5-4ecb-91e0-84f0800be253">

I have added the dependency using Swift Package Manager to both the project and pods
<img width="769" alt="image" src="https://github.com/aws-amplify/amplify-ui/assets/106042257/ffc50d44-8423-4809-b7b8-42d3892e5e0b">

<img width="778" alt="image" src="https://github.com/aws-amplify/amplify-ui/assets/106042257/08a4b697-61b6-4355-bf4b-22003035d968">

Will be great if I can have alternative on how I can add the dependency correctly to my self made React Native library, thank you.

**Comment by valentinchelle** [comment]
Any udpate on this?

**Comment by reesscot** [comment]
Hi @valentinchelle,
Thank you for your interest, but this is currently not on our roadmap.

**Comment by iransousa** [comment]
I made my own module Liveness for react native on Swift and Kotlin and its working fine, if you need some help

**Comment by joazvsoares** [comment]
> I made my own module Liveness for react native on Swift and Kotlin and its working fine, if you need some help

Hi @iransousa , I'm very interested to learn how you made it, do you have your module in github?

**Comment by sanduluca** [comment]
Hi @iransousa i am also interested

**Comment by premjogi** [comment]
> Hi @iransousa , I'm very interested to learn how you made it, do you have your module in github?

Yes @iransousa I'm also wanted to know how you implemented in RN using NativeModules.

@iransousa @sanduluca @joazvsoares

**Comment by premjogi** [comment]
> > I made my own module Liveness for react native on Swift and Kotlin and its working fine, if you need some help
> 
> Hi @iransousa , I'm very interested to learn how you made it, do you have your module in github?

@iransousa @sanduluca @joazvsoares Can you please share your REPO for the reference?

**Comment by manan-schmooze** [comment]
Any updates on this?
@iransousa if you could please share your implementation or maybe write a blog about it somewhere...?

**Comment by shuo-hiwintech** [comment]
> 我在 Swift 和 Kotlin 上为 React Native 制作了自己的模块 Liveness，它运行良好，如果你需要帮助

Bro, can you share this please?

**Comment by shuo-hiwintech** [comment]
+1

**Comment by jacoblogan** [comment]
Thank you everyone for your excitement and support for this functionality.  We will consider adding an officially supported React Native component for this feature.

**Comment by shuo-hiwintech** [comment]
> Thank you everyone for your excitement and support for this functionality. We will consider adding an officially supported React Native component for this feature.

I wonder if you would consider onAnalysisComplete / or some other way, you can get a certain video frame (or image / S3 FileId) with the highest recognition rate?
Then you can do a lot of functions based on this: e.g. face login / various recognition comparisons.

**Comment by iransousa** [comment]
Hello Guys 

Sorry guys, these months have been very busy so I was unable to provide an example or how to integrate React Native with Aws Face Liveness, this week I will post a complete functional example for Android and the other week for IOS. I can't post the original code due to the contract with the company that hired me, but the app is in production and working on Android and IOS.

I have made a repository available with the necessary files for integration with Android but I will post the complete example ok

React Native Version 0.73.9

https://github.com/iransousa/faceliveness-react-native/tree/example

**Comment by mateitudose** [comment]
That would be awesome @iransousa, a solution like that would be perfect until we get an official SDK which is confirmed to be on roadmap

**Comment by CamiloMonteroBDS** [comment]
Hello guys, thank you for your efforts in enhancing the Face Liveness feature! I’m particularly interested in using this functionality in a React Native application. Could you please share if there’s any timeline or update on when this feature might be available for React Native?

Thanks in advance!

**Comment by jordanvn** [comment]
Hi @CamiloMonteroBDS, thank you for your interest. However, this is not currently on our roadmap.

**Comment by matheus-provin** [comment]
Are there any future plans for this Amplify integration ? Would love to see a React Native SDK for Face Liveness

**Comment by iransousa** [comment]
Good afternoon!

I recently had to update my company's project from React Native 0.73.9 to 0.79.5. As a result, the AWS Liveness integration I had previously implemented for both Android and iOS in React Native stopped working.

I've since created a separate module that now handles the Android integration successfully, and I'm making it available. I'll be working on implementing the same for iOS in the coming days.

Since I had to separate it from the main application module, I can now make it available.

https://github.com/iransousa/aws-liveness-react-native

**Comment by pascalito007** [comment]
> Good afternoon!
> 
> I recently had to update my company's project from React Native 0.73.9 to 0.79.5. As a result, the AWS Liveness integration I had previously implemented for both Android and iOS in React Native stopped working.
> 
> I've since created a separate module that now handles the Android integration successfully, and I'm making it available. I'll be working on implementing the same for iOS in the coming days.
> 
> Since I had to separate it from the main application module, I can now make it available.
> 
> https://github.com/iransousa/aws-liveness-react-native


I tried to run your project and never succeed. can you please provide a more complete guide beginner friendly on how to setup everything ?
Thanks

**Comment by iransousa** [comment]
> > Good afternoon!
> > I recently had to update my company's project from React Native 0.73.9 to 0.79.5. As a result, the AWS Liveness integration I had previously implemented for both Android and iOS in React Native stopped working.
> > I've since created a separate module that now handles the Android integration successfully, and I'm making it available. I'll be working on implementing the same for iOS in the coming days.
> > Since I had to separate it from the main application module, I can now make it available.
> > https://github.com/iransousa/aws-liveness-react-native
> 
> I tried to run your project and never succeed. can you please provide a more complete guide beginner friendly on how to setup everything ? Thanks

Can you give me more information so I can help you better? What error occurred so I can improve the documentation.

Node version, Java version, Android Studio version

**Comment by pascalito007** [comment]
> > > Good afternoon!
> > > I recently had to update my company's project from React Native 0.73.9 to 0.79.5. As a result, the AWS Liveness integration I had previously implemented for both Android and iOS in React Native stopped working.
> > > I've since created a separate module that now handles the Android integration successfully, and I'm making it available. I'll be working on implementing the same for iOS in the coming days.
> > > Since I had to separate it from the main application module, I can now make it available.
> > > https://github.com/iransousa/aws-liveness-react-native
> > 
> > 
> > I tried to run your project and never succeed. can you please provide a more complete guide beginner friendly on how to setup everything ? Thanks
> 
> Can you give me more information so I can help you better? What error occurred so I can improve the documentation.
> 
> Node version, Java version, Android Studio version


I was testing on ios but it seems the implementation is not ready for ios. Can you please provide an implementation that works for ios ? the Android one works fine

Thanks for your efforts

**Comment by olawalethefirst** [comment]
Hi guys, I am excited to share with you all, that I have worked on an Expo Module for integrating Face Liveness natively in react native applications. You can check it out [here](https://github.com/olawalethefirst/expo-face-liveness-for-aws-amplify).

If you care deeply about I approached the implementation, I have written extensively about it in this [blog post](https://dev.to/olawalethefirst/how-i-bridged-aws-face-liveness-to-expo-with-a-native-module-2o49).
